### ISSUE TITLE
Bug: no-undef

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#8311]
**Tell us about your environment**

* **ESLint Version:** `v3.18.0`
* **Node Version:** `v7.7.3`
* **npm Version:** `v4.4.4`

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Config is inline in source code.

**What did you do? Please include the actual source code causing the issue.**

```js
/*eslint no-undef: "error" */
var fn2 = function fn() {};
function fn3() {fn();fn2();}
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
/tmp/eslint-no-undef-bug.js
  3:17  error  'fn' is not defined  no-undef
```

**Comment by zloirock** [comment]
It's a correct behaviour, try `fn3()`.

**Comment by ilyavolodin** [comment]
As @zloirock noted, this is correct behavior. If you try to execute your code in Node or Browser, you will get the same error about `fn` not being defined.

**Comment by alberto** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the ESLint Gitter if you have any other issues!
